### PR TITLE
Feat/cleanup session api

### PR DESCRIPTION
**Pull request by zachferland**

### CODE REVIEW

**Comment by zachferland** [comment]
partly follow up to https://github.com/ceramicstudio/js-glaze/pull/134

**Comment by zachferland** [comment]
cc @pawartur for api changes mentioned before

File: packages/did-session/src/index.ts [hunk 1]
```diff
@@ -71,52 +69,108 @@
...
+ * the domain name of your app. If you are using the library in a non web based environment you will need to pass
+ * the `domain` option otherwise an error will thrown.
+ *
+ * ```ts
+ * const session = await DIDSession.authorize(authProvider, { resources: [...], domain: 'YourAppName' })
+ * ```
+ *
+ * ## Typical usage pattern
  *
  * A typical pattern is to store a serialized session in local storage and load on use if available. Then
```
Comment (PaulLeCam) [inline packages/did-session/src/index.ts:107]: Could we add some warnings about how session management is the responsibility of the app and they should make sure to understand the security implications of storing sessions in browsers please?
Maybe we can mention that local storage is used here for illustrative purposes but not a recommended mechanism?

**Review by PaulLeCam — APPROVED** [review]
Looks great, the API changes are very nice!
Just wrote a comment about session management, I assume most apps will just use local storage but could we add some warnings about it please?

File: packages/did-session/src/index.ts (hunk 1 above)
Comment (zachferland) [reply packages/did-session/src/index.ts:107]: yeah good idea, more info on what is stored/risks is useful although idk of other implementation options (at least now) 

will add section before merging

Comment (PaulLeCam) [reply packages/did-session/src/index.ts:107]: Thanks!

**Pull request merged**